File: 0070.爬楼梯/0070-爬楼梯.rs
impl Solution {
    pub fn climb_stairs(n: i32) -> i32 {
        fn fib(n: i32) -> i32 {
            if n == 1 || n == 2 {
                return n;
            }
            let mut a = 1;
            let mut b = 2;
            let mut n = n;
            while n > 2 {
                let tmp = a;
                a = b;
                b = tmp + b;
                n -= 1;
            }
            return b;
        }
        return fib(n);
    }
}

File: 0070.爬楼梯/0070-爬楼梯.py
from functools import lru_cache


class Solution:
    def climbStairs(self, n: int) -> int:
        @lru_cache(maxsize=128)
        def fib(n):
            if n <= 2:
                return n
            return fib(n - 1) + fib(n - 2)

        return fib(n)
